File: nodejs/long-method/good/notificationService.js
class NotificationService {
    constructor(db) {
        this.db = db;
    }

    async sendWelcomeNotification(userId) {
        await this.db.query(
            'INSERT INTO notifications (user_id, type, message, created_at) VALUES ($1, $2, $3, NOW())',
            [userId, 'welcome', 'Welcome to our platform!']
        );
    }
}

class Logger {
    static logRegistration(email) {
        const logMessage = `User registered: ${email} at ${new Date().toISOString()}`;
        console.log(logMessage);
    }
}

module.exports = {
    NotificationService,
    Logger
};

File: java/long-method/good/NotificationService.java
import java.sql.Connection;

public class NotificationService {
    private Connection db;

    public NotificationService(Connection db) {
        this.db = db;
    }

    public void sendWelcomeNotification(int userId) {
        // Notification logic
        System.out.println("Sending welcome notification to user " + userId);
    }
}
